text/0026-semantic-versioning.md

```diff
@@ -0,0 +1,109 @@
+As a Buildpack/Builder user,<br/>
+I want to know how versioning is done,
+so I can use the appropriate version and
+decide when to upgrade.
+
+## Semantics
+
+Semantic Versioning 2.0.0 defines versioning syntax as follows [semver.org](https://semver.org/):
+
+> Given a version number MAJOR.MINOR.PATCH, increment the:
+> 
+>   - MAJOR version when you make incompatible API changes,
+>   - MINOR version when you add functionality in a backwards compatible manner, and
+>   - PATCH version when you make backwards compatible bug fixes.
+
+The proposal is to conform to the above semantics as much as possible.
+
+### Changes
+
+In terms of Paketo Buildpacks/Builders, this translates to:
+
+* Bumping MAJOR is analogous to an "`app or related buildpacks may not work without change`" relation between releases.
+* Bumping MINOR is analogous to an "`updates or adds features`" relation between releases.
+* Bumping PATCH is analogous to a "`fixes issue`" relation between releases.
+
+The following table lists the usual circumstances/events where bumping a particular component is expected:
+
+#### Buildpack
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| \<Any incompatible change\> | \<Feature additions but compatible with previous release\> | \<Fixes but no major feature changes\> |
+| Change in Buildpack ID | Addition/removal of dependency<br>(but default-version not changed to a version with breaking change) | Bug fixes |
+| Change in supported Stacks | Buildpack runs a new command the causes backward compatible<br>change in behavior (e.g. x-install buildpack adds running "x clean") | Performance improvement, internal tweaks |
+| Change in API (Provides/Requires) | Change of file type of build/detect executable<br>(e.g. text to binary) | CVE fixes |
+| Default version of dependency bumped to incompatible version<br>(e.g. php7 to php8) | New configuration option (e.g. new environment variable support) | Change of required build tool version to package buildpack<br>(e.g. go version) |
+| Removal of configuration options<br>(e.g. buildpack.yml) | Change in Buildpack API version |  |
+| Removal of publishing to a certain image registry<br>(e.g. gcr.io, dockerhub) | Publishing to a new image registry |  |
+|  | Change in <br>[buildpack-specific metadata](https://github.com/buildpacks/spec/blob/buildpack/v0.6/buildpack.md#build-plan-toml) |  |
+
+#### Buildpack (Language-family specific)
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| Implementation buildpack additions/removals/major bumps<br>that cause incompatible change |  |  |
+
+
+#### Builder
+| `Major`                 | `Minor`                   | `Patch`                   |
+|-------------------------|---------------------------|---------------------------|
+| Lifecycle major bumps   | Minor bumps of components | Patch bumps of components |
+| Stack Image major bumps |                           |                           |
+| Buildpack major bumps   |                           |                           |
+
```

Comment:
```suggestion
#### Stack
| `Major`                 | `Minor`                   | `Patch`                   |
|-------------------------|---------------------------|---------------------------|
| Removal of  a package   | Addition of a new package | Update of a package       |
```

@martyspiewak does this look accurate for what we would expect for stacks?

Reply:
Yup, looks right to me.

Reply:
Added. Thanks